hexrd/ui/image_file_manager.py

```diff
@@ -143,14 +143,49 @@ def is_hdf(self, extension):
 
         return False
 
+    def guess_path(self, f):
+        h5 = h5py.File(f, 'r')
+        location = ''
+        dataname='DATA'
+        if 'ATTRIBUTES' in h5:
+            if ('PINHOLE_ORIENTED_IMAGE' in h5['ATTRIBUTES']) and ('TARGET_ORIENTED_IMAGE' in h5['ATTRIBUTES']):
+                h5.close()
+                try:
+                    imageseries.open(f, 'hdf5', path='ATTRIBUTES/TARGET_ORIENTED_IMAGE/DATA', dataname=dataname)
+                except:
+                    imageseries.open(f, 'hdf5', path='ATTRIBUTES/PINHOLE_ORIENTED_IMAGE/DATA', dataname=dataname)
+            elif 'PSL_FADE_CORR_IMAGE' in h5['ATTRIBUTES']:
+                print(f'PSL_FADE_CORR_IMAGE in ATTRIBUTES')
+                location = 'PSL_FADE_CORR_IMAGE'
+            elif 'HDR_IMAGE' in h5['ATTRIBUTES']:
+                location = 'HDR_IMAGE'
+            elif 'CORR_IMAGE' in h5['ATTRIBUTES']:
+                location = 'CORR_IMAGE'
+            elif 'TIME_ADJUSTED_IMAGE' in h5['ATTRIBUTES']:
+                location = 'TIME_ADJUSTED_IMAGE'
+            elif 'PSL_IMAGE' in h5['ATTRIBUTES']:
+                location = 'PSL_IMAGE'
+        elif 'DATA' in h5:
+            location = 'DATA'
+        h5.close()
+        path=f'ATTRIBUTES/{location}/DATA'
+        imageseries.open(f, 'hdf5', path=path, dataname=dataname)
+        return path, dataname
+
     def path_exists(self, f):
         try:
             path, dataname = HexrdConfig().hdf5_path
             imageseries.open(f, 'hdf5', path=path, dataname=dataname)
             self.path = HexrdConfig().hdf5_path
             return True
         except:
-            return False
+            try:
+                path, dataname = self.guess_path(f)
+                HexrdConfig().hdf5_path = [path, dataname]
+                self.path = HexrdConfig().hdf5_path
+                return True
+            except:
+                return False
```

Comment:
```suggestion
            try:
                path, dataname = self.guess_path(f)
            except:
                return False
            else:
                HexrdConfig().hdf5_path = [path, dataname]
                self.path = HexrdConfig().hdf5_path
                return True
```
Let's minimize the section in the `try` block to whatever we are expecting to raise an exception.

We should also avoid catching bare exceptions, since there may be valid exceptions that need to be propagated to the user. We should figure out exactly what exceptions we are expecting and catch those.